### PR TITLE
Properly use the downloaded torrent file in the qBittorrent plugin (except on magnet links).

### LINKED ISSUE #1390: Qbittorrent plugin doesn't add torrent to client
### Expected behaviour:
Flexget to download the torrent and insert into qbittorrent

### Actual behaviour:
Flexget download the torrent to tmp and claims it added to qbittorrent but it never shows up in the client

### Steps to reproduce:
This is tricky as for some site the torrent gets added to the client without any problem. And in some case it doesn't, but the torrent file is created in the tmp folder.

#### Config:
```
templates:
  tv:
    qbittorrent:
      host: XXXXXX
      port: XXXX
      username: XXXX
      password: XXXX
      label: series
    series:
      settings:
        hdtv:
          quality: hdtv <720p
          set:
            path: path_to_directory/{{series_name}}
      hdtv:
        - Hit the floor
tasks:
  dlseries:
    headers:
      Cookie: "XXXXXXXX"
    rss: FEED URL
    verify_ssl_certificates: no
    template: tv

```
  
#### Log:
```
2016-09-06 17:39 DEBUG    series        dlseries           Series accepting. <TORRENTNAME> meets quality hdtv
2016-09-06 17:39 VERBOSE  task          dlseries           ACCEPTED: `<TORRENTNAME>` by series plugin because target quality
2016-09-06 17:39 DEBUG    urlrewriter   dlseries           Checking 1 entries
2016-09-06 17:39 INFO     download      dlseries           Downloading: <TORRENTNAME>
2016-09-06 17:39 DEBUG    download      dlseries           Downloading url '<TORRENTURL>'
2016-09-06 17:39 DEBUG    utils.requests dlseries           Fetching <TORRENTURL>
2016-09-06 17:39 DEBUG    download      dlseries           <TORRENTNAME> field file set to: /tmp/tmpX_QHcL/87a3ce82ae44e8d94daa57e28191bcb9
2016-09-06 17:39 DEBUG    download      dlseries           filename header latin1 decoded
2016-09-06 17:39 DEBUG    download      dlseries           Found filename from headers: <TORRENTNAME>
2016-09-06 17:39 DEBUG    download      dlseries           Mimetype guess for application/x-bittorrent is ['.torrent']
2016-09-06 17:39 DEBUG    download      dlseries           Filename <TORRENTNAME>.torrent extension matches to mime-type
2016-09-06 17:39 DEBUG    download      dlseries           Finishing download_entry() with filename <TORRENTNAME>.torrent
2016-09-06 17:39 DEBUG    download      dlseries           Successfully retrieved <TORRENTNAME> from <TORRENTURL>
2016-09-06 17:39 VERBOSE  details       dlseries           Summary - Accepted: 1 (Rejected: 0 Undecided: 39 Failed: 0)
2016-09-06 17:39 DEBUG    modif_torrent dlseries           <TORRENTNAME> seems to be a torrent
2016-09-06 17:39 DEBUG    torrent_files dlseries           <TORRENTNAME> files: [u'<TORRENTNAME>.mkv', u'Sample/<TORRENTNAME>.mkv', u'<TORRENTNAME>.nfo']
2016-09-06 17:39 DEBUG    torrent_size  dlseries           <TORRENTNAME> size: 319.150249481 MB
2016-09-06 17:39 DEBUG    qbittorrent   dlseries           Successfully connected to qBittorrent
2016-09-06 17:39 DEBUG    qbittorrent   dlseries           Added task to qBittorrent

```

### Additional information:

- Flexget Version: 2.3.18
- Python Version: 2.7.9
- Installation method: pip
- OS and version: Debian 8.5 (OSMC)
- Link to crash log:

Please verify that the following data is present before submitting your issue:

-Paste or link to a paste service ([pastebin](http://pastebin.com/) for example) of relevant config (preferably full config including templates if present. Remember to redact any personal information!
- Paste or link to a paste service of debug level logs of relevant task/s. Use `flexget -L debug execute --tasks <Task_name>`
- Flexget version (Use `flexget -V` to get it).
- Full Python version (`2.7.11` for example). Run `python -V` to get it.
- Installation method (pip, git install, etc.)
- OS and version
- Attach crash log if available

### PR DESCRIPTION
**Pull request by dsemi**
### Motivation for changes:
Make qbittorrent more like the other client plugins.
### Detailed changes:

Allow qBittorrent plugin to properly handle input torrent files and magnet links.
Also merge [change](https://github.com/Flexget/Flexget/pull/984) to set 'label' as 'category' since qBittorrent decided to change the API and the original PR looks abandoned.

### Addressed issues:

- Fixes #1390

### CODE REVIEW

**Comment by liiight** [comment]
I'm gonna test this later today

File: flexget/plugins/output/qbittorrent.py [hunk 1]
```diff
@@ -60,10 +61,20 @@ def connect(self, config):
...
 
-    def add_torrent(self, data):
+    def add_torrent_file(self, filepath, data):
         if not self.connected:
             raise plugin.PluginError('Not connected.')
         multipart_data = {k: (None, v) for k, v in data.items()}
+        with open(filepath, 'rb') as f:
+            multipart_data['torrents'] = f
+            self.session.post(self.url + '/command/upload', files=multipart_data)
+        log.debug('Added task to qBittorrent')
```
Comment (cvium) [inline flexget/plugins/output/qbittorrent.py]: That doesn't sound right.

File: flexget/plugins/output/qbittorrent.py [hunk 2]
```diff
@@ -77,6 +88,46 @@ def prepare_config(self, config):
...
 
+    def add_entries(self, task, config):
+        for entry in task.accepted:
+            formdata = {}
+
+            savepath = entry.get('path', config.get('path'))
+            if savepath:
+                formdata['savepath'] = savepath
+
+            label = entry.get('label', config['label']).lower()
```
Comment (cvium) [inline flexget/plugins/output/qbittorrent.py]: Why .lower()?

File: flexget/plugins/output/qbittorrent.py [hunk 3]
```diff
@@ -77,6 +88,46 @@ def prepare_config(self, config):
...
+            if label:
+                formdata['label'] = label # qBittorrent v3.3.3-
+                formdata['category'] = label # qBittorrent v3.3.4+
+
+            downloaded = not entry['url'].startswith('magnet:')
```
Comment (cvium) [inline flexget/plugins/output/qbittorrent.py]: I'd probably rename this to `is_magnet` or something.

File: flexget/plugins/output/qbittorrent.py [hunk 4]
```diff
@@ -77,6 +88,46 @@ def prepare_config(self, config):
...
+                log.info('    Label: %s', formdata.get('label'))
+                continue
+
+            if downloaded:
+                if 'file' not in entry:
+                    entry.fail('File missing?')
+                    continue
+                if not os.path.exists(entry['file']):
+                    tmp_path = os.path.join(task.manager.config_base, 'temp')
+                    log.debug('entry: %s' % entry)
```
Comment (cvium) [inline flexget/plugins/output/qbittorrent.py]: Logger module can handle string formatting, so change the % to a comma. You already did it correctly in the --test block.

**Comment by cvium** [comment]
I took the liberty of re-inserting the template you deleted. Please fill it with the proper information. It's there to help us.

File: flexget/plugins/output/qbittorrent.py (hunk 1 above)
Comment (dsemi) [inline flexget/plugins/output/qbittorrent.py]: @cvium Can you be more specific?

Comment (cvium) [inline flexget/plugins/output/qbittorrent.py]: You don't add a task, you add entries. The debug message is pointless as it provides no information.

**Comment by dsemi** [comment]
Made the changes.

File: flexget/plugins/output/qbittorrent.py [hunk 5]
```diff
@@ -96,32 +96,32 @@ def add_entries(self, task, config):
...
                 formdata['savepath'] = savepath
 
-            label = entry.get('label', config['label']).lower()
+            label = entry.get('label', config.get('label'))
             if label:
                 formdata['label'] = label # qBittorrent v3.3.3-
                 formdata['category'] = label # qBittorrent v3.3.4+
 
-            downloaded = not entry['url'].startswith('magnet:')
+            is_magnet = not entry['url'].startswith('magnet:')
```
Comment (cvium) [inline flexget/plugins/output/qbittorrent.py]: You should negate the expression if you want to call it `is_magnet`.

**Comment by cvium** [comment]
@dsemi I assume you've tested it?

File: flexget/plugins/output/qbittorrent.py [hunk 6]
```diff
@@ -77,6 +88,46 @@ def prepare_config(self, config):
...
+                formdata['label'] = label # qBittorrent v3.3.3-
+                formdata['category'] = label # qBittorrent v3.3.4+
+
+            is_magnet = entry['url'].startswith('magnet:')
+
+            if task.manager.options.test:
+                log.info('Test mode.')
+                log.info('Would add torrent to qBittorrent with:')
+                if not is_magnet:
+                    log.info('    File: %s', entry.get('file'))
```
Comment (paranoidi) [inline flexget/plugins/output/qbittorrent.py]: We don't use empty space prefixes / indentations in log messages.

**Comment by paranoidi** [comment]
Looks pretty good

File: flexget/plugins/output/qbittorrent.py (hunk 6 above)
Comment (cvium) [inline flexget/plugins/output/qbittorrent.py]: Not entirely true. It's used in other places for indentation.

**Comment by dsemi** [comment]
@cvium Yes

@paranoidi Removed the spaces.

**Comment by liiight** [comment]
For what it's worth, tested in master and seem to work well. 
Thanks for this!

**Comment by dsemi** [comment]
No problem :)

**Pull request merged**